src/agent0/core/base/agent/policy_agent.py

```diff
@@ -1,107 +0,0 @@
-
-Policy = TypeVar("Policy", bound=BasePolicy)
-MarketInterface = TypeVar("MarketInterface")
-MarketAction = TypeVar("MarketAction")
-
-
-class PolicyAgent(LocalAccount, Generic[Policy, MarketInterface, MarketAction]):
-    r"""Enact policies on smart contracts and tracks wallet state"""
-
-    def __init__(self, account: LocalAccount, initial_budget: FixedPoint | None = None, policy: Policy | None = None):
-        """Initialize an agent and wallet account
-
-        Arguments
-        ---------
-        account: LocalAccount
-            A Web3 local account for storing addresses & signing transactions.
-        initial_budget: FixedPoint | None, optional
-            The initial budget for the wallet bookkeeping.
-        policy: Policy | None, optional
-            Policy for producing agent actions.
-            If None, then a policy that executes no actions is used.
-
-        Note
-        ----
-        If you wish for your private key to be generated, then you can do so with:
-
-        .. code-block:: python
-
-            >>> from eth_account.account import Account
-            >>> from agent0.core.base import PolicyAgent
-            >>> agent = PolicyAgent(Account().create("CHECKPOINT_BOT"))
-
-        Alternatively, you can also use the Account api to provide a pre-generated key:
-
-        .. code-block:: python
-
-            >>> from eth_account.account import Account
-            >>> from agent0.core.base import PolicyAgent
-            >>> agent = PolicyAgent(Account().from_key(agent_private_key))
-
-        The PolicyAgent has the same properties as a Web3 LocalAgent.
-        For example, you can get public and private keys as well as the address:
-
-            .. code-block:: python
-
-                >>> address = agent.address
-                >>> checksum_address = agent.checksum_address
-                >>> public_key = agent.key
-                >>> private_key = bytes(agent)
-
-        """
-        if policy is None:
-            self.policy = NoActionPolicy(NoActionPolicy.Config())
-        else:
-            self.policy = policy
-
-        # TODO budget should have a flag to allow for "the budget is however much this wallet has"
-        # https://github.com/delvtech/agent0/issues/827
-        if initial_budget is None:
-            initial_budget = FixedPoint(0)
-
-        # State variable defining if this agent is done trading
-        self.done_trading = False
-        super().__init__(account._key_obj, account._publicapi)  # pylint: disable=protected-access
-
-        self.wallet = EthWallet(
-            address=HexBytes(self.address),
-            balance=Quantity(amount=initial_budget, unit=TokenType.BASE),
-        )
-
-    @property
-    def checksum_address(self) -> ChecksumAddress:
-        """Return the checksum address of the account."""
-        return Web3.to_checksum_address(self.address)
-
-    def get_trades(self, interface: MarketInterface) -> list[Trade[MarketAction]]:
```

Comment:
We need a `preview_agent_trades` endpoint to complement `execute_agent_trades` that allows the user to see what the policy would suggest without actually doing it. This supports human-in-the-loop behavior.

Reply:
Will implement in a future pr.
https://github.com/delvtech/agent0/issues/1491